### ISSUE TITLE
Error submitting or deleting items - URI too long when user is in a large number of groups

### ISSUE THREAD

**Issue by ddean4040** [DSpace/DSpace#9544]
**Describe the bug**
When submitting or deleting an item, a user who belongs to a large number of groups but is not an admin will get an error message and the operation will fail.

The REST service logs:
```
2024-05-01 14:32:57,281 INFO  ... ... org.dspace.app.rest.utils.DSpaceAPIRequestLoggingFilter @ Before request [GET /server/api/discover/search] originated from /mydspace?configuration=workspace
2024-05-01 14:32:57,300 ERROR ... ... org.dspace.app.rest.exception.DSpaceApiExceptionControllerAdvice @ An exception has occurred (status:500)
org.apache.solr.client.solrj.impl.HttpSolrClient$RemoteSolrException: Error from server at http://solr:8983/solr/search: Expected mime type application/octet-stream but got text/html. <h1>Bad Message 414</h1><pre>reason: URI Too Long</pre>
        at org.apache.solr.client.solrj.impl.HttpSolrClient.executeMethod(HttpSolrClient.java:635) ~[solr-solrj-8.11.2.jar:8.11.2 ... - mdrob - 2022-06-13 11:27:56]
        at org.apache.solr.client.solrj.impl.HttpSolrClient.request(HttpSolrClient.java:266) ~[solr-solrj-8.11.2.jar:8.11.2 ... - mdrob - 2022-06-13 11:27:56]
        at org.apache.solr.client.solrj.impl.HttpSolrClient.request(HttpSolrClient.java:248) ~[solr-solrj-8.11.2.jar:8.11.2 ... - mdrob - 2022-06-13 11:27:56]
        at org.apache.solr.client.solrj.SolrRequest.process(SolrRequest.java:214) ~[solr-solrj-8.11.2.jar:8.11.2 ... - mdrob - 2022-06-13 11:27:56]
        at org.apache.solr.client.solrj.SolrClient.query(SolrClient.java:1003) ~[solr-solrj-8.11.2.jar:8.11.2 ... - mdrob - 2022-06-13 11:27:56]
        at org.apache.solr.client.solrj.SolrClient.query(SolrClient.java:1018) ~[solr-solrj-8.11.2.jar:8.11.2 ... - mdrob - 2022-06-13 11:27:56]
        at org.dspace.content.EntityTypeServiceImpl.getSubmitAuthorizedTypes(EntityTypeServiceImpl.java:151) ~[dspace-api-7.6.1.jar:7.6.1]
```

It looks like the query in EntityTypeServiceImpl.java is using GET instead of POST (it does not specify `solrSearchCore.REQUEST_METHOD`). We were able to workaround the issue for now by increasing the max header size in Solr.

This is on DSpace 7.6.1.

**To Reproduce**
1. Add a user to a large number of groups (our user is in 550 groups) but not the Administrator group
2. Try to submit or delete an item

**Expected behavior**
Operation completes successfully.

**Related work**
Looks similar to this old bug: #6732

**Comment by tdonohue** [comment]
@ddean4040 : Thanks for the additional details.  Based on those details, I've determined this is a *new issue* (not a duplicate).

The core problem appears to be in `EntityTypeServiceImpl.getSubmitAuthorizedTypes()` (which is used by both of the endpoints you mentioned).  That method appears to append **every group** you are a member of to a (potentially very large) Solr query.  See the code at: https://github.com/DSpace/DSpace/blob/main/dspace-api/src/main/java/org/dspace/content/EntityTypeServiceImpl.java#L134-L139

So, this appears to be a very inefficient query, especially if you are a member of a large number of groups. 

Pulling this over to our 8.0 board in search of a volunteer.

**Comment by tdonohue** [comment]
@ddean4040 : I just realized your analysis in the description is 100% correct.  This problem should be fixed if we send this query to Solr via a `POST` instead of a `GET`.  In other areas of DSpace, we are primarily using `POST` queries.

I'll create a small PR which should fix this.

**Comment by tdonohue** [comment]
Small PR which should fix the issue #9548.   @ddean4040 : If you have a chance to verify/test this small fix, please do let us know your findings.

**Comment by alanorth** [comment]
I hadn't noticed these, but we do have a large repository with a number of users who belong to many groups. Sure enough, I see over a thousand in the past month or so (logs are zstd compressed):

```console
$ zstdgrep -a 'URI Too Long' log/dspace.log-2024-04-* | wc -l
1423
```

I will find time to test this patch as well.
